src/main/java/net/sf/marineapi/nmea/event/AbstractSentenceListener.java

```diff
@@ -101,9 +101,15 @@ public void readingStopped() {
 	@SuppressWarnings("unchecked")
 	public void sentenceRead(SentenceEvent event) {
 		Sentence sentence = event.getSentence();
-		Class<?>[] interfaces = sentence.getClass().getInterfaces();
-		if (Arrays.asList(interfaces).contains(this.expectedType)) {
-			sentenceRead((T) sentence);
+		if (this.expectedType instanceof Class<?>) {
```

Comment:
If I have this right, every class is an instance of Class<?> (e.g. String.class is Class<String> etc), so I think the `expectedType` could be of type Class<?>, cast immediately in constructor, thus reducing the triggering logic into simple

`if (this.expectedType.isAssignableFrom(sentence.getClass())) {
    sentenceRead((T) sentence);
}`